pinot-query-runtime/src/main/java/org/apache/pinot/query/mailbox/GrpcSendingMailbox.java

```diff
@@ -66,13 +67,19 @@ public void send(TransferableBlock block)
     }
     Preconditions.checkState(!_statusObserver.isFinished(), "Mailbox: %s is already closed", _id);
     _contentObserver.onNext(toMailboxContent(block));
+    _isEarlyTerminated = _statusObserver.isFinished();
```

Comment:
This will almost always throw exception on line 68

Reply:
removed